File: samples/transfer_image/sender.py
import sys, os
sys.path.insert(0, os.path.abspath('..'))

from client import *
from utils.image_utils import *
from utils.aes_utils import *

sharedKey = "LGWlhb329Y09oluI"
imagePathToSend = "sample.jpg"

# Register device
print "Registering device..."
msg = "SHARE #pubkey KEY @senz #time {} ^dev1 signature".format(getTimestamp())
sendMessage(msg)

# Convert image to byte string
byteString = imageToString(imagePathToSend)

# Encrypt using AES Crypto
aes = AESUtils(sharedKey)
byteString = aes.encrypt(byteString)

# Send the message
print "Press enter to send the image..."
raw_input()
msg = "DATA $image {} @dev2 #time {} ^dev1 signature\n".format(byteString, getTimestamp())
sendMessage(msg)

File: senz-client-samples/javascript/transfer_image/sender.js
const AESUtil=require("../utils/aes_utils");
const imageUtil=require("../utils/image_utils");
const logger = require("../utils/winstonConfig");
const client=require("../client");
const express=require("express");
const app=express();
const port=process.env.port || 3001;
const sharedKey=[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16];
//Register Device 
logger.info("Registering Device");
var time=client.getTimestamp();
var regmsg=`SHARE #pubkey KEY @senz #time ${time} ^dev1 signature\n`;
const imagePathToSend="sample.jpg";
//Convert image to byte string
var byteString=imageUtil.imageToString(imagePathToSend);
//Encrypt using AES Crypto
var aes=new AESUtil.AESUtils(sharedKey);
var byteString=aes.encrypt(byteString);
//logger.info(byteString);
//Send Message
var senmsg = `DATA $image ${byteString} @dev2 #time ${time} ^dev1 signature\n`
//sendImage(regmsg,senmsg);
var registeringDevice=function(regmsg)
{
    client.sendMessage(regmsg).then(function(registered){
        logger.info(registered);
    })
}
var sendingMessage=function(senmsg){
    client.sendMessage(senmsg).then(function(senData){
        logger.info(senData);
        client.sendMessage(`UNSHARE #pubkey KEY @senz #time ${time} ^dev1 signature\n`).then(function(registered){
            logger.info(registered);
        })
    })
}

// client.sendMessage(regmsg).then(function(registered){
//     logger.info(registered);
//     client.sendMessage(senmsg).then(function(sentData){
//         logger.info(sentData);
        
//     })
// });
app.get("/",function(req,res){
    res.sendFile("sendRegister.html",{root:__dirname});
})
app.get("/reg1",function(req,res){
    registeringDevice(regmsg);
    res.sendFile("sendSend.html",{root:__dirname})
})
app.get("/sen1",function(req,res){
    sendingMessage(senmsg);
    res.send("File Sent !!");
})
app.listen(port,function(err){
    if(err)
    throw err;
    logger.info(`server running on port ${port}`);
})
